charts/templates/statefulset.yaml

```diff
@@ -21,6 +21,60 @@ spec:
       affinity: {{- toYaml . | nindent 8}}
       {{- end }}
       terminationGracePeriodSeconds: {{ .Values.terminationGracePeriod }}
+      {{- if ne .Values.bootstrapHostName "" }}
+      initContainers: 
+      - name: configure-group
+        image: "{{ .Values.initContainerImage.repository }}:{{ .Values.initContainerImage.tag }}"
+        command:
+          - sh
+          - '-c'
+          - |
+            log () {
+                      local TIMESTAMP=$(date +"%Y-%m-%d %T.%3N")
+                      echo "${TIMESTAMP}  $@"
+                    }
+            log "Info: [initContainer] Begin init container configure-group execution"
+            if [[ $POD_NAME != *-0 ]]; then
+              log "Info: [initContainer] Skipping creation of group $MARKLOGIC_GROUP as $POD_NAME is not the first node in the group."
+            else
+              HOST_RESP_CODE=`curl --anyauth -m 20 -s -o /dev/null -w "%{http_code}" -X GET http://${MARKLOGIC_BOOTSTRAP_HOST}:8002/manage/v2/hosts --anyauth --user ${MARKLOGIC_ADMIN_USERNAME}:${MARKLOGIC_ADMIN_PASSWORD}`
+              if [[ ${HOST_RESP_CODE} -ne 200 ]]; then
+                log "Error: [initContainer] Bootstrap host $MARKLOGIC_BOOTSTRAP_HOST not found."
+              else
+                GROUP_CFG_TEMPLATE='{"group-name":"%s"}'
+                GROUP_CFG=$(printf "$GROUP_CFG_TEMPLATE" "$MARKLOGIC_GROUP")
+                GROUP_RESP_CODE=`curl --anyauth -m 20 -s -o /dev/null -w "%{http_code}" -X GET http://${MARKLOGIC_BOOTSTRAP_HOST}:8002/manage/v2/groups/${MARKLOGIC_GROUP} --anyauth --user ${MARKLOGIC_ADMIN_USERNAME}:${MARKLOGIC_ADMIN_PASSWORD}`
+
+                if [[ ${GROUP_RESP_CODE} -eq 200 ]]; then
+                  log "Info: [initContainer] Skipping creation of group $MARKLOGIC_GROUP as it already exists on the MarkLogic cluster." 
+                else
+                  log "Info: [initContainer] Group $MARKLOGIC_GROUP does not exist, creating $MARKLOGIC_GROUP on the MarkLogic cluster."
+                  curl --anyauth --user ${MARKLOGIC_ADMIN_USERNAME}:${MARKLOGIC_ADMIN_PASSWORD} -m 20 -s -X POST -d "${GROUP_CFG}" -H "Content-type: application/json" http://${MARKLOGIC_BOOTSTRAP_HOST}:8002/manage/v2/groups
```

Comment:
What happens if this fails for some reason?

Reply:
If this commends fails, the execution will continue and still logs "init container execution completed". This should be fixed.
We could add a retry logic. If it still fails, log the error and exit 1 fail the whole deployment so that the customer can inspect the issue and redeploy again.